lib/Functional/Functional.ark

```diff
@@ -1,17 +1,18 @@
 {
+
+    (import "../Math/Arithmetic.ark")
+
     (let compose (fun (f g)
         (fun (y &f &g) (f (g y)))))
     
     (let zip (fun (a b) {
-        (if (!= (len a) (len b)) nil
-            {
-                (mut c [])
-                (mut idx 0)
-                (while (< idx (len a)) {
-                    (set c (append c [(@ a idx) (@ b idx)]))
-                    (set idx (+ 1 idx))})
-                c
-                })
+        (let m (min (len a) (len b) ))
+        (mut c [])
+        (mut idx 0)
+        (while (< idx m) {
+            (set c (append c [(@ a idx) (@ b idx)]))
+            (set idx (+ 1 idx))})
+        c
```

Comment:
since it changed, you should rewrite the corresponding unit test, in tests/list-tests.ark